app/scripts/proactive/templates/job-variable-template.html

```diff
@@ -139,11 +139,27 @@ <h4 class="modal-title"><%- jobName %> </h4>
                 <label class="short" data-toggle="tooltip" title="<%- model %>"><%- shortenModel %></label>
                 <button title="Browse, Upload User Space files, and Select one folder" value="<%- key %>" class="var-userfolder-button btn far fa-folder-open" type="button" style="background-color:Transparent"></button>
             <% } else { %>
+                <% if(resolvedModel && ( (resolvedModel.toLowerCase().indexOf('pa:hidden') == -1) ) ) { %>
                 <textarea class="variableValue textareavalues submitVariableValue" rows=1 name="<%- variable.Name %>" data-variable-model="<%- model %>" id="<%- key %>" ><%- variable.Value %></textarea>
-                <label class="short" data-toggle="tooltip" title="<%- model %>"><%- referenceType %></label>
-                <% if( referenceType === "PA:GLOBAL_FILE" ) { %>
+                <% } else  { %>
+                <input class="variableValue textareavalues submitVariableValue" name="<%- key %>" type="password" data-variable-model="<%- model %>" id="<%- key %>" value="<%- variable.Value %>" autocomplete="new-password" />
+                <% } %>
+                <label class="short" data-toggle="tooltip" title="<%- model %>"><%- resolvedModel %></label>
+                <% if( resolvedModel && resolvedModel.toLowerCase().indexOf('pa:global_file') != -1 ) { %>
                 <button title="Browse Global Space files, Upload, and Select one file" value="<%- key %>" class="var-globalfile-button btn far fa-folder-open" type="button" style="background-color:Transparent"></button>
                 <% } %>
+                <% if(resolvedModel && resolvedModel.toLowerCase().indexOf('pa:user_file') != -1 ) { %>
```

Comment:
Why not combining these conditions with the previous ones for model like `if( (model && model.toLowerCase().indexOf('HANDLING_MODEL')!=-1) || (resolvedModel  && resolvedModel.toLowerCase().indexOf('HANDLING_MODEL')!=-1) )` to reduce the duplicated code for a specific model's display.
p.s. I think you can also create a function to check whether `model` or `resolvedModel` matches a specific handling model.

Reply:
Maybe you could have added some comments to explain the conditions